sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -74,6 +74,15 @@ trait CheckAnalysis extends PredicateHelper {
     }
   }
 
+  private def getNumInputFileBlockSources(operator: LogicalPlan): Int = {
+    operator match {
+      case _: LeafNode => 1
```

Comment:
Unable to check it in `CheckAnalysis`. Both `HadoopRDD` and `FileScanRDD` have the same issues. To block both, we need to add the check as another rule.